base/inference.jl

```diff
@@ -1221,16 +1232,58 @@ function findlabel(labels, l)
 end
 
 function label_counter(body)
-    l = 0
+    l = -1
     for b in body
         if isa(b,LabelNode) && (b::LabelNode).label > l
             l = (b::LabelNode).label
         end
     end
-    l
+    return l
 end
 genlabel(sv) = LabelNode(sv.label_counter += 1)
 
+function find_gensym_uses(body)
+    uses = Vector{Int}[]
+    for line = 1:length(body)
+        find_gensym_uses(body[line], uses, line)
+    end
+    return uses
+end
+function find_gensym_uses(e::ANY, uses, line)
+    if isa(e,GenSym)
+        id = (e::GenSym).id+1
```

Comment:
i'm not sure what you want there. they are a bit annoying, but they save me from doing more `-1`'s in the C code. i could write an accessor function, but that seems a bit overkill. it's simply the unfortunate downside to breaking with the C-style zero-based indexing